### PR TITLE
kaufkauflist: 2.2.0 → 3.0.0

### PR DESCRIPTION
**Pull request by auroraanna**
## Description of changes

- Updated kaufkauflist
- added update script
- formatting improvements

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by auroraanna** [comment]
Not sure if this is major enough for the release notes…

> If you removed packages or made some major NixOS changes, write about it in the release notes for the next stable release in [`nixos/doc/manual/release-notes`](./nixos/doc/manual/release-notes).
>
> --`CONTRIBUTING.md`

sounds different from

> (Package updates) Added a release notes entry if the change is major or breaking
>
> --the issue template

**Comment by donovanglover** [comment]
Release: https://codeberg.org/annaaurora/kaufkauflist/releases/tag/v3.0.0
Diff: https://codeberg.org/annaaurora/kaufkauflist/compare/v2.2.0...v3.0.0

**Review by donovanglover — COMMENTED** [review]
- Update script should support Gitea now. See: https://github.com/Mic92/nix-update/pull/134
- Remove `buildPackages` and replace `buildPackages.buildGoModule` / `buildPackages.esbuild` with the ones already in the attribute set

**Comment by auroraanna** [comment]
> * Remove `buildPackages` and replace `buildPackages.buildGoModule` / `buildPackages.esbuild` with the ones already in the attribute set

It's needed for crosscompilation, see <https://github.com/NixOS/nixpkgs/pull/208964#discussion_r1068801499>.

**Comment by auroraanna** [comment]
Though I removed `esbuild` and `buildGoModule` because they weren't used.

File: pkgs/applications/misc/kaufkauflist/default.nix [hunk 1]
```diff
@@ -1,31 +1,31 @@
...
     repo = "kaufkauflist";
     rev = "v${version}";
-    hash = "sha256-a7C4yHTHPhL5/p1/XsrMA0PnbIzer6FShDiwUMOg69Y=";
+    hash = "sha256-W/FlHLZYYG/s9NdAqm0OJHlpTZtEG4iaegc4iOnAwWk=";
   };
 
-  npmDepsHash = "sha256-uQ4XoaR3JjvPm8EQ2pnDM+x4zjVn4PEHq7BRqVbvFyw=";
+  npmDepsHash = "sha256-d1mvC72ugmKLNStoemUr8ISCUYjyo9EDWdWUCD1FMiM=";
 
   ESBUILD_BINARY_PATH = "${lib.getExe esbuild'}";
```
Comment (donovanglover) [inline pkgs/applications/misc/kaufkauflist/default.nix:30]:
```suggestion
  ESBUILD_BINARY_PATH = lib.getExe esbuild';
```
Nit from https://github.com/NixOS/nixpkgs/pull/231904#discussion_r1200980641

**Review by donovanglover — APPROVED** [review]
LGTM

**Pull request merged**